library/get_ip.py

```diff
@@ -4,63 +4,63 @@
 import requests
 import json
 from library.common import ProvisionerError
-from library.common import get_machine_by_name
+from library.common import *
 from urllib.parse import urljoin
 from urllib.parse import quote
 
 
 class IPGetter(object):
-    def __init__(self, mrpurl, mrptoken, machine_name, interface_name='eth1'):
-        self.mrp_url = mrpurl
-        self.mrp_token = mrptoken
+    def __init__(self, requester, machine_name, interface_name='eth1'):
+        self.requester = requester
         self.interface = interface_name
         self.machine_id = get_machine_by_name(
-            self.mrp_token, self.mrp_url, machine_name)['id']
+            self.requester, machine_name)['id']
         self.machine_ip = ''
 
     def get_interfaces(self):
-        headers = {'Authorization': self.mrp_token}
-        url = urljoin(self.mrp_url,
-                      "/api/v1/machine/{}/interface".format(self.machine_id))
-        r = requests.get(url, headers=headers)
-        if r.status_code != 200:
-            raise ProvisionerError('Error fetching {} {}, HTTP {} {}'.format(url, self.interface, r.status_code,
-                                                                             r.reason))
-        if len(r.json()) == 0:
-            raise ProvisionerError(
-                'Error no machine with id "{}"'.format(self.machine_id))
+        path = "/api/v1/machine/{}/interface".format(self.machine_id)
+        try:
+            result = self.requester.get(path)
+        except Exception as err:
+            raise err
 
-        return r.json()
+        return result
 
     def get_ip(self):
```

Comment:
Can you common up the interface check loop into a local method? and then use it on all other methods.

Reply:
This is now the URLhandler::get_field_by_interface method.